src/program.tsx

```diff
@@ -49,7 +49,7 @@ class Program extends React.Component<ProgramProps, ProgramState> {
     console.log("[Gongfu] Running Update for ", msg.tag);
 
     const { update } = this.props;
-    const { model, effect } = update(msg, this.state.model);
+    const [model, effect] = update(msg, this.state.model);
```

Comment:
Looks good to me, and it will be easier to get the result of the `update` function when there are name conflicts:
we can simply do `const [homeModel, homeEffect] = update(...)`
instead of using the ES6 destructuring syntax with a rename `const {model: homeModel, effect: homeEffect} = update(...)`

Reply:
Can the effect be made optional for updates that don't have a side effect?

Reply:
@newyankeecodeshop only the top level `update` function right now is required to return an Effect.